core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -1301,9 +1301,12 @@ private class BufferReleasingInputStream(
       block
     } catch {
       case e: IOException if detectCorruption =>
-        val diagnosisResponse = checkedInOpt.map { checkedIn =>
-          iterator.diagnoseCorruption(checkedIn, address, blockId)
-        }
+        val diagnosisResponse = checkedInOpt
+          // TODO SPARK-36284 Add shuffle checksum support for push-based shuffle
+          .filterNot { _ => blockId.isShuffleChunk }
+          .map { checkedIn =>
+            iterator.diagnoseCorruption(checkedIn, address, blockId)
+          }
```

Comment:
Shall we do the check inside `diagnoseCorruption()` to ensure other calls on it are also safe?

Reply:
Updated